core/test/financial-templates/Liquidatable.js

```diff
@@ -273,12 +282,72 @@ contract("Liquidatable", function(accounts) {
           .toString()
       );
     });
-    it("Liquidation amount not greater than or equal to the underlying collateral in the contract", async () => {
-      const invalidAmount = amountOfCollateralToLiquidate.sub(toBN(toWei("149.99")));
-      assert(
-        await didContractThrow(
-          liquidationContract.createLiquidation(sponsor, { rawValue: invalidAmount.toString() }, { from: liquidator })
-        )
+    it("Partial liquidation", async () => {
+      // Request a withdrawal.
+      const withdrawalAmount = amountOfSynthetic.divn(5);
+      await liquidationContract.requestWithdrawal({ rawValue: withdrawalAmount.toString() }, { from: sponsor });
+
+      // Position starts out with `amountOfSynthetic` tokens.
+      const expectedLiquidatedTokens = amountOfSynthetic.divn(5);
+      const expectedRemainingTokens = amountOfSynthetic.sub(expectedLiquidatedTokens);
+
+      // Position starts out with `amountOfCollateral` collateral.
+      const expectedLockedCollateral = amountOfCollateral.divn(5);
+      const expectedRemainingCollateral = amountOfCollateral.sub(expectedLockedCollateral);
+      const expectedRemainingWithdrawalRequest = withdrawalAmount.sub(withdrawalAmount.divn(5));
+
+      // Create partial liquidation.
+      let uuid = await liquidationContract.createLiquidation.call(
+        sponsor,
+        { rawValue: pricePerToken.toString() },
+        { rawValue: amountOfSynthetic.divn(5).toString() },
+        { from: liquidator }
+      );
+      await liquidationContract.createLiquidation(
+        sponsor,
+        { rawValue: pricePerToken.toString() },
+        { rawValue: amountOfSynthetic.divn(5).toString() },
+        { from: liquidator }
+      );
+      let position = await liquidationContract.positions(sponsor);
+      let liquidation = await liquidationContract.liquidations(sponsor, uuid);
+      assert.equal(expectedRemainingTokens.toString(), position.tokensOutstanding.toString());
+      assert.equal(expectedRemainingWithdrawalRequest.toString(), position.withdrawalRequestAmount.toString());
+      assert.equal(
+        expectedRemainingCollateral.toString(),
+        (await liquidationContract.getCollateral(sponsor)).toString()
+      );
+      assert.equal(expectedLiquidatedTokens.toString(), liquidation.tokensOutstanding.toString());
+      assert.equal(expectedLockedCollateral.toString(), liquidation.lockedCollateral.toString());
+
+      // A independent and identical liquidation can be created.
+      await liquidationContract.createLiquidation(
+        sponsor,
+        // Due to rounding problems, have to increase the pricePerToken.
+        { rawValue: pricePerToken.muln(2).toString() },
```

Comment:
Can you explain the rounding case here? Will help me to think about rounding in general

Reply:
You end up liquidating slightly less collateral than you expect due to, e.g., (1/3 = 0.33). That leaves the remaining position slightly overcollateralized than what it was when it started, so for subsequent liquidations you need to pass in a slightly higher token price. We could tweak the math to produce different rounding patterns if we want. What do you think?

Reply:
I think you shouldn't worry about rounding stuff during these tests, maybe just acknowledge the potential to produce them?

Reply:
To get that assert to pass below, had to accept the rounding. Happy to do it another way if possible, I'd also rather than not deal with any rounding at all here.